app/src/main/java/de/westnordost/streetcomplete/screens/main/bottom_sheet/MoveNodeFragment.kt

```diff
@@ -0,0 +1,207 @@
+        return listener?.getMapPositionAt(screenPos)
+    }
+
+    private fun onClickOk() {
+        val pos = getPosition() ?: return
+        if (!checkIsDistanceOkAndUpdateText(pos)) return
+        viewLifecycleScope.launch {
+            val action = MoveNodeAction(pos)
+            elementEditsController.add(editType, node, ElementPointGeometry(node.position), "survey", action)
+            listener?.onMovedNode(editType, pos)
+        }
+    }
+
+    override fun onClickMapAt(position: LatLon, clickAreaSizeInMeters: Double) = false
+
+    @UiThread override fun onMapMoved(position: LatLon) {
+        if (checkIsDistanceOkAndUpdateText(position))
+            okButtonContainer.popIn()
+        else
+            okButtonContainer.popOut()
+    }
+
+    private fun checkIsDistanceOkAndUpdateText(position: LatLon): Boolean {
+        val moveDistance = position.distanceTo(node.position)
+        return when {
+            moveDistance < MIN_MOVE_DISTANCE -> {
+                binding.titleLabel.setText(R.string.node_moved_not_far_enough)
+                false
+            }
+            moveDistance > MAX_MOVE_DISTANCE -> {
+                binding.titleLabel.setText(R.string.node_moved_too_far)
+                false
+            }
+            else -> {
+                binding.titleLabel.text = resources.getString(R.string.node_moved,displayUnit.format(moveDistance.toFloat()))
+                true
+            }
+        }
+    }
+
+    @UiThread override fun onClickClose(onConfirmed: () -> Unit) {
+        if (!hasChanges) {
+            onConfirmed()
+        } else {
+            activity?.let {
+                AlertDialog.Builder(it)
+                    .setMessage(R.string.confirmation_discard_title)
+                    .setPositiveButton(R.string.confirmation_discard_positive) { _, _ ->
+                        onConfirmed()
+                    }
+                    .setNegativeButton(R.string.short_no_answer_on_button, null)
+                    .show()
+            }
+        }
+    }
+
+    companion object {
+        private const val ARG_NODE = "node"
+        private const val ARG_QUEST_TYPE = "quest_type"
+
+        fun create(elementEditType: ElementEditType, node: Node): MoveNodeFragment {
+            val f = MoveNodeFragment()
+            f.arguments = bundleOf(
+                ARG_NODE to Json.encodeToString(node),
+                ARG_QUEST_TYPE to elementEditType.name
+            )
+            return f
+        }
+    }
+}
+
+// Require a minimum distance for two reasons:
+// 1. The map is not perfectly precise, especially displayed road widths may be off by a few meters,
+//     so it may be hard to tell whether something really is misplaced (e.g. bench along a path)
+//     without good aerial imagery.
+// 2. The value added by moving nodes by such small distance, even if correct, is rather low.
+private const val MIN_MOVE_DISTANCE = 2.0
```

Comment:
A third good reason would be:

What is the size of a (e.g. shop) icon shown at maximum zoom in meters? Basically, the larger the icon, the less sense it makes to move it by some centimeters. So if a hydrant icon is displayed at e.g. 2x2 meters (I don't know), the position imprecision is already about 1m because it is not really possible for the user to ascertain exactly what is the center of the icon. The icons are not shown on a pin after all, like they are in iD
![grafik](https://user-images.githubusercontent.com/4661658/202709052-a8406393-d3e1-4352-b334-a6f997a71350.png)

Reply:
I was thinking about that, but this probably depends on the device. For tablet the size might be much different from a phone, and even between phones this could be different.
On my phone I would estimate shop icons are a bit larger than 2x2 m, but not by much.

Reply:
Why would this depend on the device? The icons on the map are scaled, too. I.e. on devices with a low DPI, the icons should appear at the same size as on devices with a high DPI.

Reply:
I wasn't aware the map scales together with the icons

Reply:
So the size is the same everywhere